dist/formatters/no-internal-summary-table-creator.js

```diff
@@ -0,0 +1,126 @@
+    const rowStrings = splitRows.map(row => {
+        const numLines = Math.max(...row.map(cellLines => cellLines.length));
+        const lines = Array(numLines).fill().map((_, i) =>
+            '|' + row.map((cellLines, j) => ' ' + (cellLines[i] || '').padEnd(colWidths[j]) + ' ').join('|') + '|'
+        );
+        return lines.join('\n');
+    });
+
+    // Join everything together
+    return rowSeparator + '\n' + rowStrings.join('\n' + rowSeparator + '\n') + '\n' + rowSeparator;
+}
+
+  messages.forEach((message) => {
+    const errorMessage = message.text ?? message.message;
+    // regex pattern for error message for no internal violations
+    const re = new RegExp('(.*) "(.*)" is (.*).');
+    const match = errorMessage.match(re);
+    if (match) {
+      const kind = match[1];
+      const name = match[2];
+      const tag = match[3];
+      const filePath = message.location?.file ?? message.filePath;
+      const relativeFilePath = filePath.replace(cwdWithSeparator, '');
+      const problemFileMapKey = `${relativeFilePath};;${tag}`;
+      const fileDetails = problemFiles.get(problemFileMapKey) ?? { locations: "" , count: 0};
+      const newLocations = fileDetails.locations + `${message.location?.line ?? message.line}:${message.location?.column ?? message.column},`;
+      problemFiles.set(problemFileMapKey, { locations: newLocations , count: fileDetails.count + 1});
+
+      const key = `${kind};;${name};;${tag}`;
+      const currentCount = errorTracker.get(key) ?? 0;
+      const newCount = currentCount + 1;
+      errorTracker.set(key, newCount);
+
+      const tagViolationCount = tagViolationsTracker.get(tag) ?? 0;
+      tagViolationsTracker.set(tag, tagViolationCount + 1);
+    }
+  });
+
+    if (problemFiles.size === 0 || errorTracker.size === 0)
+      return;
+
+    // iterate over the problemFiles and for each value, remove the last comma from locations
+    for (const [key, value] of problemFiles.entries()) {
+      const locations = value.locations.slice(0, -1);
+      problemFiles.set(key, { locations, count: value.count});
+    }
+
+    const filesSummaryRows = [
+      ['File', 'Locations', 'Tag', '# of Occurrences'],
+      ...Array.from(problemFiles.entries()).map(([filePath, { locations, count }]) => {
+        const [relativeFilePath, tag] = filePath.split(';;');
+        return [relativeFilePath, locations, tag, count];
+      })
+    ];
+    const filesSummaryTable = createAsciiTable(filesSummaryRows, maxCellWidth);
+
+    const rows = [
+      ["Kind and Name", "Tag", "# of Occurrences"],
+      ...Array.from(errorTracker.entries()).map(([key, value]) => {
+        const [kind, name, tag] = key.split(";;");
+        return [`${kind} ${name}`, tag, value];
+      })
+
+    ];
+    const summaryTable = createAsciiTable(rows, maxCellWidth);
+
+    const tagViolationsRows = [
+      ["Tag", "# of Occurrences"],
+      ...Array.from(tagViolationsTracker.entries()).map(([tag, value]) => [tag, value]),
```

Comment:
this `...Array.from(x.entries()).map...` pattern is used a few times... maybe extract a function?

Reply:
Since each .map() implementation is slightly different in each instance, even if they are refactored into a single function a separate map function would still need to be passed to that refactored function, so let's leave it as is.

Github copilot agrees with me too 😄:
![image](https://github.com/iTwin/eslint-plugin/assets/98850418/4e17b088-8352-4445-b3dc-6cbeb4ed87ac)